vcpkg/contributing/maintainer-guide.md

```diff
@@ -215,11 +215,11 @@ Examples:
 - The Azure SDKs (of the form `azure-Xxx`) have a `public-preview` feature.
 - `imgui` has an `experimental-docking` feature which engages their preview docking branch which uses a merge commit attached to each of their public numbered releases.
 
-### Default features should enable behaviors, not APIs
+### <a name="default-features-should-enable-behaviors-not-apis"></a> Default features must not add APIs
 
-If a consumer is depending directly upon a library, they can list out any desired features easily (`library[feature1,feature2]`). However, if a consumer _does not know_ they are using a library, they cannot list out those features. If that hidden library is like `libarchive` where features are adding additional compression algorithms (and thus behaviors) to an existing generic interface, default features offer a way to ensure a reasonably functional transitive library is built even if the final consumer doesn't name it directly.
+If depending directly upon a library, one can list out any desired features easily (`library[feature1,feature2]`). However, if a consumer _does not know_ they are using a library, they cannot know to list features. Default features are intended to ensure that a reasonably functional transitive library is built even if the final consumer doesn't name it directly. For example, `libarchive`'s features add additional compression algorithms (and thus behaviors) to an existing generic interface, and customers would likely find `libarchive` with no selected compression algorithms unusable.

```

Comment:
```suggestion
Default features are intended to ensure that a reasonably functional build of a library gets installed when a consumer doesn't request any features explicitly. For example, `libarchive` exposes features that enable compression algorithms (and thus behaviors) to an existing generic interface; if built without any of such features, the library has no utility. A secondary problem that default features solve, is that transitive consumers don't need to know the features of such a library to get a usable build.

A downside of default features is that they are difficult for transitive users to disable, and a user may not know they consume a library that installs default features at all.
```

I think our preamble paragraph needs a bit of work, here's my suggestion but I'm not totally convinced with the wording.

Reply:
I somewhat disagree that they are 'primary' and 'secondary' concerns, or that they are unrelated like this. The situation of 'generic interface with plugins where at least one plugin is expected' results in those being interrelated.

Reply:
I removed the "secondary" part and instead added a bit about why we consider disabling default features difficult. In hope that it dissuades more contributors from adding them.

Reply:
This suggestion still looses explaining that the lack of knowledge is the reason default features behave the way they do. I'm also not a fan of the stronger statement "has no utility": we don't know why someone might be requesting no features. Clearly we think there might be hypothetical utility there, or we wouldn't allow disabling default features in the first place.

I've tried to incorporate some of your suggestions here, can you check again?